x-pack/plugins/synthetics/public/apps/synthetics/hooks/use_last_x_checks.ts

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useMemo } from 'react';
+import { useSelector } from 'react-redux';
+import { useEsSearch, createEsParams } from '@kbn/observability-plugin/public';
+import { Ping } from '../../../../common/runtime_types';
+
+import {
+  EXCLUDE_RUN_ONCE_FILTER,
+  getLocationFilter,
+  SUMMARY_FILTER,
+} from '../../../../common/constants/client_defaults';
+import { selectServiceLocationsState } from '../state';
+import { useSyntheticsRefreshContext } from '../contexts/synthetics_refresh_context';
+import { SYNTHETICS_INDEX_PATTERN, UNNAMED_LOCATION } from '../../../../common/constants';
+
+export function useLastXChecks<Fields>({
+  monitorId,
+  locationId,
+  fields = ['*'],
+  size = 50,
+}: {
+  monitorId: string;
+  locationId: string;
+  fields?: string[];
+  size?: number;
+}) {
+  const { lastRefresh } = useSyntheticsRefreshContext();
+  const { locationsLoaded, locations } = useSelector(selectServiceLocationsState);
+
+  const params = createEsParams({
+    index: locationsLoaded ? SYNTHETICS_INDEX_PATTERN : '', // don't run query until locations are loaded
+    body: {
+      size,
+      query: {
+        bool: {
+          filter: [
+            SUMMARY_FILTER,
+            EXCLUDE_RUN_ONCE_FILTER,
+            {
+              term: {
+                config_id: monitorId,
+              },
+            },
+          ],
+          ...getLocationFilter({
+            locationId,
+            locationName:
+              locations.find((location) => location.id === locationId)?.label || UNNAMED_LOCATION,
+          }),
+        },
+      },
+      sort: [{ '@timestamp': 'desc' }],
+      fields,
+    },
+  });
+
+  const { data } = useEsSearch<Ping, typeof params>(params, [lastRefresh], {
+    name: 'getLast50MonitorRunsByLocation',
+  });
+
+  const dataAsJSON = JSON.stringify(data?.hits?.hits);
```

Comment:
i don't think this makes sense. Why not just use data in dependency array?

Reply:
I wanted to make sure that, if the data is refetched on `lastRefresh` but the content remains the same, it wouldn't return data from the below useMemo and rerender the metric item with brand new data. Wdyt? How can I avoid this?

Reply:
but es call will still happen as lastRefresh is dependency of `useEsSearch`

Reply:
Yes, but I'm trying to avoid `Metric` component from rerendering when it doesn't receive new data.

Reply:
I should probably try to prevent re-rendering of Metric component at the Metric component level rather than here.

Reply:
I think this strategy will work as @dominiqueclarke is saying, I've employed this tactic before, too. I think the `stringify` doesn't slow down the function too much for the data volume we should be processing and it will skip making a new object and re-rendering. If it's worth the complexity idk; how expensive is that render?